smdebug/tensorflow/base_hook.py

```diff
@@ -126,60 +128,82 @@ def _get_worker_name(self) -> str:
         It is safe to return the CONFIG_DEFAULT_WORKER_NAME in this case.
         :return: str
         """
-        try:
+        assert self.distribution_strategy is not None
```

Comment:
When would this be the case? We may want to throw an error in the constructor (earlier on) rather than waiting until this point.

Reply:
This happens when we call this method before calling get_distribution_strategy. The distribution strategy is initialized to None, as it's not available during hook construction. Later when the graph is ready, that's when correct distribution strategy is returned. 
In current code, I call get_distribution_strategy before the methods which use them. These asserts are just safety checks if someone changes the order of these methods. Will add this comment